### PR TITLE
Adding more logging APIs

### PR DESCRIPTION
**Pull request by jasonsandlin**

### CODE REVIEW

File: Include/httpClient/httpProvider.h [hunk 1]
```diff
@@ -65,6 +65,7 @@ STDAPI HCHttpCallSetContext(
     _In_ void* context
     ) HC_NOEXCEPT;
 
+
 /////////////////////////////////////////////////////////////////////////////////////////
```
Comment (matt-vanderkolk) [inline Include/httpClient/httpProvider.h]: Extra line intentional?

File: Source/HTTP/httpcall.cpp [hunk 2]
```diff
@@ -127,7 +127,7 @@ HRESULT perform_http_call(
                         }
                         catch (...)
                         {
-                            HC_TRACE_ERROR(HTTPCLIENT, "HCHttpCallPerform [ID %llu]: failed", static_cast<HC_CALL*>(call)->id);
+                            if (call->traceCall) HC_TRACE_ERROR(HTTPCLIENT, "HCHttpCallPerform [ID %llu]: failed", static_cast<HC_CALL*>(call)->id);
```
Comment (matt-vanderkolk) [inline Source/HTTP/httpcall.cpp]: >HC_TRACE_ERROR [](start = 49, length = 14)

Not wrapping this in braces puts an assumption that this is always going to be a 1-line Macro. Is that guaranteed?

Comment (matt-vanderkolk) [reply Source/HTTP/httpcall.cpp]: Here and elsewhere

---
In reply to: [181256747](https://github.com/Microsoft/libHttpClient/pull/128#discussion_r181256747) [](ancestors = 181256747)

File: Include/httpClient/httpProvider.h (hunk 1 above)
Comment (jasonsandlin) [reply Include/httpClient/httpProvider.h]: yep.  i love me some extra lines before the //// 

---
In reply to: [181256500](https://github.com/Microsoft/libHttpClient/pull/128#discussion_r181256500) [](ancestors = 181256500)

File: Source/Logger/trace_internal.h [hunk 3]
```diff
@@ -2,5 +2,191 @@
...
+// HC_TRACE_WARNING_ENABLE [0,1]
+// HC_TRACE_IMPORTANT_ENABLE [0,1]
+// HC_TRACE_INFORMATION_ENABLE [0,1]
+// HC_TRACE_VERBOSE_ENABLE [0,1]
+//
+// these should not be manually modified, they are automatically set based on
+// the values of HC_TRACE_BUILD_LEVEL
+
+#if HC_TRACE_BUILD_LEVEL > HC_PRIVATE_TRACE_LEVEL_OFF
+#define HC_TRACE_ENABLE 1
```
Comment (matt-vanderkolk) [inline Source/Logger/trace_internal.h]: >HC_TRACE_ENABLE [](start = 8, length = 15)

Sorry, we also use these

**Pull request merged**